.github/workflows/check-pl-matrix.yaml

```diff
@@ -0,0 +1,136 @@
+name: Check PennyLane Compatibility
+
+on:
+  push:
+    branches: [ main ]
+  workflow_dispatch:
+  pull_request:
+
+jobs:
+  check-matrix:
+    name: Build Compatibility Matrix against PL/Lightning
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        catalyst: ['latest', 'release']
+        pennylane: ['latest', 'release']
+        lightning: ['latest', 'release']
+
+    steps:
+    - name: Cancel previous runs
+      uses: styfle/cancel-workflow-action@0.11.0
+      with:
+        access_token: ${{ github.token }}
+
+    - name: Checkout Catalyst repo
+      uses: actions/checkout@v3
+    - if: ${{ matrix.catalyst == 'release' }}
+      run: |
+        git checkout $(git tag | sort -V | tail -1)
+
+    - name: Get LLVM version
+      id: llvm-hash
+      run: echo "llvm-hash=$(grep llvm .dep-versions | awk -F '=' '{ print $2 }')" >> $GITHUB_OUTPUT
+
+    - name: Get MHLO version
+      id: mhlo-hash
+      run: echo "mhlo-hash=$(grep mhlo .dep-versions | awk -F '=' '{ print $2 }')" >> $GITHUB_OUTPUT
+
+    - name: Install deps
+      run: |
+        sudo apt-get install -y cmake ninja-build ccache clang lld libomp-dev
+        python3 -m pip install -r requirements.txt
+
+    - name: Get Catalyst Build Dependencies (latest)
+      uses: actions/cache@v3
+      with:
+        path: qir-stdlib-build
+        key: Linux-qir-stdlib-build
+        fail-on-cache-miss: True
+    - uses: actions/cache@v3
+      with:
+        path: mlir/llvm-project
+        key: Linux-llvm-${{ steps.llvm-hash.outputs.llvm-hash }}-default-source
+        enableCrossOsArchive: True
+        fail-on-cache-miss: True
+    - uses: actions/cache@v3
+      with:
+        path: llvm-build
+        key: ${{ runner.os }}-llvm-${{ steps.llvm-hash.outputs.llvm-hash }}-default-build
+        fail-on-cache-miss: True
+    - uses: actions/cache@v3
+      with:
+        path: mhlo-build
+        key: ${{ runner.os }}-mhlo-${{ steps.mhlo-hash.outputs.mhlo-hash }}-default-build
+        fail-on-cache-miss: True
+    - uses: actions/cache@v3
+      with:
+        path: .ccache
+        key: ${{ runner.os }}-ccache-${{ github.run_id }}
+        restore-keys: ${{ runner.os }}-ccache-
+
+
+    - name: Build Lightning Runtime (latest)
+      if: ${{ matrix.lightning == 'latest' }}
+      run: |
+        COMPILER_LAUNCHER="" \
```

Comment:
The runtime build usually takes less than 2min, I want to say not much gains from caching. On the other hand, the latest version (build against master) of lightning tends to change more often that other dependencies which are cached, so maybe caching doesn't bring any additional benefits.

Reply:
If you are talking about compiler caching (what `COMPILER_LAUNCHER=""` deactivates), I suppose we could do it to speed up the runtime build time, but here the completion speed of the action is not so important. In order to do it we would have to cache the ccache cache however, which would be another step in the pipeline and produce more cache entries.

If you are talking about caching the build itself, that is not possible since we want to get the latest version of the code.

Reply:
@dime10 talking about `COMPILER_LAUNCHER`. Cool, no worries!